docarray/array/chunk.py

```diff
@@ -31,12 +28,7 @@ def __init__(self, docs, reference_doc: 'Document'):
         """
         self._ref_doc = reference_doc
         super().__init__(docs)
-        if (
-            isinstance(
-                docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
-            )
-            and self._ref_doc is not None
-        ):
+        if isinstance(docs, (Iterable, Iterator)) and self._ref_doc is not None:
```

Comment:
Iterator is also a subclass of Iterable so we can just keep Iterable here
```suggestion
        if isinstance(docs, Iterable) and self._ref_doc is not None:
```